src/rayon_impls.rs

```diff
@@ -0,0 +1,119 @@
+use crate::HashMap;
+use rayon::iter::{FromParallelIterator, IntoParallelIterator, ParallelExtend, ParallelIterator};
+use std::hash::{BuildHasher, Hash};
+
+impl<K, V, S> ParallelExtend<(K, V)> for HashMap<K, V, S>
+where
+    K: Clone + Hash + Ord + Send + Sync + 'static,
+    V: Send + Sync + 'static,
+    S: BuildHasher + Sync,
+{
+    // This is of limited use due to the `&mut self` parameter. See `par_extend_sync`
+    fn par_extend<I>(&mut self, par_iter: I)
```

Comment:
It's still of use in that it can be filled concurrently for _this_ extend, even if it requires exclusive access up front.

Another possibility is to `impl ParallelExtend for &HashMap`, so the `&mut self` is really `&mut &HashMap`. You could do that in addition to this base implementation, replacing `par_extend_sync`, although I'm not sure off-hand how well auto-ref/deref will let that play out.

Still, you do want that base implementation so things like `partition` and `unzip` can work.

Reply:
Yeah, I more meant "this is not something the library user will usually want to use directly." I think almost everyone who uses `flurry` wraps it in an `Arc` immediately. You're right it is important to have it of course. (I removed the comment.)

Nice call with the `impl` for `&HashMap`! I feel smarter having learned this trick